src/core/simulation_results.jl

```diff
@@ -1,616 +1,133 @@
-    variable = (collect(keys(references[stage])))
-    dual = _find_duals(variable)
-    param = _find_params(collect(keys(references[stage])))
-    variable = setdiff(variable, vcat(param, dual))
-    time_stamp = DataFrames.DataFrame(Range = Dates.DateTime[])
-    time_length = sim_output.chronologies[stage]
-
-    for l in 1:length(variable)
-        date_df = references[stage][variable[l]]
-        variables[(variable[l])] = DataFrames.DataFrame()
-        for (ix, time) in enumerate(date_df.Date)
-            file_path = date_df[ix, :File_Path]
-            var = read_arrow_file(file_path)
-            variables[(variable[l])] = vcat(variables[(variable[l])], var[1:time_length, :])
-            if l == 1
-                time_stamp = vcat(time_stamp, _read_time(file_path, time_length))
-                check_file_integrity(dirname(file_path))
-            end
-        end
-    end
-    time_stamp[!, :Range] = convert(Array{Dates.DateTime}, time_stamp[!, :Range])
-    file_path = dirname(references[stage][variable[1]][1, :File_Path])
-    optimizer = read_json(joinpath(file_path, "optimizer_log.json"))
-    obj_value = Dict{Symbol, Any}(:OBJECTIVE_FUNCTION => optimizer["obj_value"])
-    param_values = _read_references(params, param, stage, references, time_length)
-    duals = _read_references(duals, dual, stage, references, time_length)
-    return SimulationResults(
-        base_power,
-        variables,
-        obj_value,
-        optimizer,
-        time_stamp,
-        duals,
-        results_folder,
-        param_values,
-    )
-end
-
-"""
-    check_file_integrity(path::String)
-
-Checks the hash value for each file made with the file is written with the new hash_value to verify the file hasn't been tampered with since written
-
-# Arguments
-- `path::String`: this is the folder path that contains the results and the check.sha256 file
-"""
-function check_file_integrity(path::String)
-    matched = true
-    for file_info in read_file_hashes(path)
-        filename = file_info["filename"]
-        expected_hash = file_info["hash"]
-        actual_hash = compute_sha256(filename)
-        if expected_hash != actual_hash
-            @error "hash mismatch for file" filename expected_hash actual_hash
-            matched = false
-        end
-    end
-
-    if !matched
-        throw(IS.HashMismatchError("The hash value in the written files does not match the read files, results may have been tampered."))
+    if execution === nothing
+        execution = maximum([parse(Int, f) for f in readdir(path) if occursin(r"^\d+$", f)])
+    end
+    execution_path = joinpath(path, string(execution))
+    if !isdir(execution_path)
+        error("Execution $execution not in the simulations results")
+    end
+    if !check_folder_integrity(execution_path)
+        @warn("The results folder $(execution_path) is not consistent with the default folder structure. This can lead to unwanted to errors or unwanted results")
```

Comment:
```suggestion
        @warn("The results folder $(execution_path) is not consistent with the default folder structure. This can lead to errors or unwanted results")
```